src/Lachain.Core/Network/BlockSynchronizer.cs

```diff
@@ -296,32 +298,31 @@ private void BlockSyncWorker()
                     const int maxBlocksToRequest = 10;
 
                     var peers = _peerHeights
-                        .Where(entry => entry.Value >= maxHeight)
-                        .Select(entry => entry.Key)
+                        .Where(entry => entry.Value > myHeight)
                         .OrderBy(_ => rnd.Next())
                         .Take(maxPeersToAsk)
                         .ToArray();
 
                     var leftBound = myHeight + 1;
-                    var rightBound = Math.Min(maxHeight, myHeight + maxBlocksToRequest);
-                    Logger.LogTrace($"Sending query for blocks [{leftBound}; {rightBound}] to {peers.Length} peers");
                     foreach (var peer in peers)
                     {
+                        var rightBound = Math.Min(peer.Value, myHeight + maxBlocksToRequest);
+                        Logger.LogTrace($"Sending query for blocks [{leftBound}; {rightBound}] to peer {peer.Key.ToHex()}");
                         _networkManager.SendTo(
-                            peer, _networkManager.MessageFactory.SyncBlocksRequest(leftBound, rightBound),
+                            peer.Key, _networkManager.MessageFactory.SyncBlocksRequest(leftBound, rightBound),
                             NetworkMessagePriority.PeerSyncMessage
                         );
                     }
 
-                    var waitStart = TimeUtils.CurrentTimeMillis();
-                    while (true)
+                    var waitForBlockExecution = 4000;
```

Comment:
Why do we use 4 secs here?

Reply:
it was hardcoded 5 secs, but assuming we wait 1 sec (max) for reply, I reduced to 4 secs. should I change it to 5 secs?

Reply:
Our block times are set in config,  I think in this case we should read it from it here (now block time is 4 secs,  for example)

Reply:
Okay, but we sending request for 10 blocks ( max ). Is it okay? Or do we need to change it?
Depending on the txes in a block, execution time may vary. 4 secs could not be enough sometimes. Do we need to think about it for now?

Reply:
ok, let use this vaule for now,  but add the task about measuring this time and adjusting it in the future

Reply:
Ok